rust/agents/validator/src/validator.rs

```diff
@@ -122,6 +124,7 @@ impl BaseAgent for Validator {
             agent_metrics,
             chain_metrics,
             core_metrics: metrics,
+            staking_config: settings.staking_config,
```

Comment:
imo instead of storing a config in the Validator that you can then lookup
```
let service_manager = self
            .staking_config
            .service_managers
            .get(&avs_domain)
```
it'd be nicer to just do that lookup in here when building the Validator once, and then store the `service_manager` as the config in the Validator